default/python/universe_generation/galaxy.py

```diff
@@ -15,19 +15,23 @@ def __init__(self, universe_width):
         self.max_dist = universe_tables.MAX_STARLANE_LENGTH
         self.cell_size = min(max(universe_width / 50, self.min_dist), self.max_dist/sqrt(2))
         self.width = int(universe_width / self.cell_size) + 1
-        self.grid = defaultdict(list)
-        print "Adjacency Grid: width", self.width, ", cell size", self.cell_size
+        self.grid = defaultdict(set)
+        print "Adjacency Grid: width {}, cell size {}".format(self.width, self.cell_size)
 
     def cell(self, pos):
+        """Returns cell index"""
         return (int(pos[0] / self.cell_size), int(pos[1] / self.cell_size))
 
     def insert_pos(self, pos):
-        self.grid[self.cell(pos)].append(pos)
+        """Inserts pos"""
+        self.grid[self.cell(pos)].add(pos)
```

Comment:
The value is the set containing pos, not pos itself.  

The `AdjacencyGrid` provides coarse chunking of the universe so that you can compare positions of systems with other systems in their vicinity without searching the entire galaxy of systems.  So given the position you can always find the set containing that position and nearby systems.